### PR TITLE
Feat/faucet page

### PR DESCRIPTION
**Pull request by oceanByte**
Depends on https://github.com/deltaDAO/mvg-portal/pull/554

## Proposed Changes

  - Add dedicated Faucet Page
  - Request tokens for known accounts
  - Support for Testnet and Devnet

### CODE REVIEW

File: src/components/Faucet/index.tsx [hunk 1]
```diff
@@ -99,59 +107,91 @@ export default function FaucetPage({
...
+    faucetTokenRequest
   ])
 
   useEffect(() => {
-    const fetchAddress = async () => {
+    const fetchAddressAndNetwork = async () => {
       try {
         const provider = new ethers.providers.Web3Provider(window.ethereum)
         const signer = provider.getSigner()
         const address = await signer.getAddress()
```
Comment (moritzkirstein) [inline src/components/Faucet/index.tsx:119]: suggestion: we can utilize wagmi react hooks here to be consistent with the repository

see [useAccount](https://wagmi.sh/react/api/hooks/useAccount) / [useNetwork](https://1.x.wagmi.sh/react/hooks/useNetwork)

File: src/components/Faucet/index.tsx [hunk 2]
```diff
@@ -99,59 +107,91 @@ export default function FaucetPage({
...
+    }
+
+    return () => {
+      if (window.ethereum) {
+        window.ethereum.removeListener(
+          'accountsChanged',
+          fetchAddressAndNetwork
+        )
+        window.ethereum.removeListener('chainChanged', fetchAddressAndNetwork)
+      }
```
Comment (moritzkirstein) [inline src/components/Faucet/index.tsx:142]: suggestion: we should rather make use of wagmi react hooks here

see [useAccount](https://wagmi.sh/react/api/hooks/useAccount) / [useNetwork](https://1.x.wagmi.sh/react/hooks/useNetwork)

File: src/components/Faucet/index.tsx [hunk 3]
```diff
@@ -99,59 +107,91 @@ export default function FaucetPage({
...
+          <li>
+            You will receive a confirmation once the tokens are successfully
+            added to your wallet.
+          </li>
+        </ol>
+        <p>
+          <strong>Note:</strong> You can only request tokens once every 12
+          hours. If you encounter any issues, please try again later.
+        </p>
+      </div>
```
Comment (moritzkirstein) [inline src/components/Faucet/index.tsx:176]: suggestion: to be consistent in the repo we should move the content of this page to its own `content/.json` file
If this is blocking an earlier release we should create an issue to track progress for the task.

File: src/pages/faucet.tsx [hunk 4]
```diff
@@ -1,23 +1,12 @@
...
-      <Page
-        title={content.title}
-        description={content.description}
-        uri={router.route}
-      >
-        <Verify didQueryString={did as string} />
-      </Page>
-    </AssetProvider>
+    <Page title={content.title} description={content.description} uri="">
+      <Verify />
```
Comment (moritzkirstein) [inline src/pages/faucet.tsx:9]: issue: rename import to `Faucet`

**Review by moritzkirstein — CHANGES_REQUESTED** [review]

File: src/components/Faucet/index.tsx (hunk 1 above)
Comment (oceanByte) [reply src/components/Faucet/index.tsx]: This will be addressed in a subsequent pull request.

File: src/components/Faucet/index.tsx (hunk 2 above)
Comment (oceanByte) [reply src/components/Faucet/index.tsx:142]: This will be addressed in a subsequent pull request.

File: src/components/Faucet/index.tsx (hunk 3 above)
Comment (oceanByte) [reply src/components/Faucet/index.tsx]: This will be addressed in a subsequent pull request.

File: src/pages/faucet.tsx (hunk 4 above)
Comment (oceanByte) [reply src/pages/faucet.tsx:9]: This will be addressed in a subsequent pull request.

**Pull request merged**